### PR TITLE
Elisp updates

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/ghc-mod/default.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { lib
-, pkgs
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/ghc-mod/default.nix:2]: @jian-lin fixup here

**Comment by AndersonTorres** [comment]
Closing, since it makes no sense to push it onto master.

**Review by jian-lin — COMMENTED** [review]
Maybe it is better to use 4 commits instead of 8 commits for this update.

**Comment by AndersonTorres** [comment]
P.S.: this should receive updates whenever Emacs is updated.

**Comment by jian-lin** [comment]
> P.S.: this should receive updates whenever Emacs is updated.

Sounds like a good idea given than nowadays Emacs update goes into the staging branch.

**Comment by jian-lin** [comment]
I have not reviewed any Elisp update PR.  What should I do during the review? (cc @adisbladis)

**Review by jian-lin — APPROVED** [review]
I have verified that the diff is the same as my run of `update-from-overlay`.

**Comment by adisbladis** [comment]
> I have not reviewed any Elisp update PR. What should I do during the review? (cc @adisbladis)

Here is what I usually do:

- Check the nix-community hydra

That should cover most outright build failures.

- Rebuild your system with the updated package set

Since Emacs updates started going to staging this is harder to do.
I'd cherry-pick relevant commits on top of unstable to test.

**Comment by AndersonTorres** [comment]
I don't like to backport elisp updates because they are automated, and my brain cells melt when a huge conflict arises.

**Comment by jian-lin** [comment]
I guess the main reason elisp updates do not happen on the stable release (NixOS 23.11) is that they can introduce breaking changes which are [not allowed][1].

[1]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#changes-acceptable-for-releases

**Pull request merged**